src/Enzyme.jl

```diff
@@ -178,7 +178,7 @@ Enzyme.autodiff(ReverseWithPrimal, x->x*x, Active(3.0))
     ModifiedBetween = Val(falses_from_args(Val(1), args...))
 
     tt    = Tuple{map(T->eltype(Core.Typeof(T)), args′)...}
-    world = GPUCompiler.codegen_world_age(Core.Typeof(f.val), tt)
+    world = GPUCompiler.tls_world_age() #(Core.Typeof(f.val), tt).min_world
```

Comment:
I don't think this replacement is equivalent here. I think we need to import the codegen_world_age that was dropped here https://github.com/JuliaGPU/GPUCompiler.jl/commit/3de799afb1e145a72aaa30dc7c29222d32802693

Reply:
Which, relatedly, can we split off the codegen_world_age refit into its own PR (which presumably should add support for both GPUCompiler 0.19 and 0.20), from the other dependency bumps?

Reply:
So PR1 is GPUCompiler bump and codegen_world_age and PR2 is rest of the stuff?

Reply:
Yeah, [though the rest may also make sense to split off, depending on how invasive the LLVM stuff is]